added interest feature in order to motivate customers to make deposits

JennyAcg opened the pull request:
Changes made in this PR:

I have implemented a interest rate feature so that for every 5 deposits of an amount greater than or equal to $20 the customer makes, the customer will receive an interest of 1.5%. Having this feature in our bank can increase customer loyalty and serve as an incentive for customers to continue using our bank. 

`src/main/java/net/testudobank/MvcController.java`

- Modified `submitDeposit()` where `NumDepositsForInterest` will be incremented using `setCustomerNumberOfDepositsForInterest()` if and only if the customer deposited an amount that is greater than or equal to $20, and the user is not in overdraft or has no money in their account.
- Modified `applyInterest()` so that each time when a customer allocates a total of 5 deposits, interest rate will be applied to their current balance and a new row will be added to the transaction table indicating this change. After the previous changes are made, `NumDepositsForinterest` will be reset to zero. 

`src/test/java/net/testudobank/tests/MvcControllerIntegTest.java`

- Added `testSimpleApplyInteresRate()` in order to test the basic functionality of the interest feature. The test verifies that interest rate will be applied to the user's balance after 5 valid deposits and that no interest will be applied to the user's balance before the 5th deposit or after the 5th deposit.
- Added `testInterestRateLessThanMinimumDepositAmt()` to verify that interest rate will not be applied to the user's balance if the user makes 5 deposits of an amount that is less than $20.

Future changes to be made:

- Update `python-sql-scripts/addCustomers.py` to allow changes in the transaction history table to be shown on the database.

s1ddh4rthc commented:
Hey Jenny, nice job on your PR comment, as well as your commit messages. Your integ test and your logic for the `applyInterest()` function look good, just make sure to finish your integ tests by next Friday.

JennyAcg commented:
Changes made in this PR:

Added two more test cases in order to test out the edge cases such as when the user enters an amount that is very close to our required amount but is still less than $20.

`src/test/java/net/testudobank/tests/MvcControllerIntegTest.java`

- Added `testInterestRateWhereDepositAmtIsLessThanAndCloseToRequiredAmount()` in order to test that the feature does not apply interest rate to the user's balance when the user did 5 deposits of $19.98.
- Added `testInterestRateWhereDepositAmtIsGreaterThanAndCloseToRequiredAmount()` in order to test that the feature applies interest rate to the user's balance when the user did 5 deposits of $20.01.
